gene2phenotype_project/gene2phenotype_app/serializers/user.py

```diff
@@ -130,31 +153,147 @@ def create(self, validated_data):
+            return user 
+
+    class Meta:
+        model = User
+        fields = ['username', 'email', 'first_name', 'last_name', 'password', 'password2', 'is_superuser', 'is_staff']
+        extra_kwargs = {'password': {'write_only': True, 'min_length': 6, 'max_length': 20}, 
+                        'email': {
+                            'validators': [
+                                UniqueValidator(
+                                    queryset=User.objects.all(),
+                                    message="This email is already registered"
+                                )
+                            ]
+                        }
+                        }
+        
+class ChangePasswordSerializer(serializers.ModelSerializer): 
+    old_password = serializers.CharField(max_length=20, min_length=6, style={'input_type' : 'password'}, write_only=True)
+    password = serializers.CharField(max_length=20, min_length=6, style={'input_type': 'password'}, write_only=True)
+    password2 = serializers.CharField(max_length=20, min_length=6, style={'input_type': 'password'}, write_only=True)
+
+    def validate(self, attrs):
+        user = self.context.get('user')
+        old_password = attrs.get('old_password')
+        if user.check_password(old_password) is False:
+            raise serializers.ValidationError({'message' : "Password incorrect, Input present password to change password"})
+        password = attrs.get('password')
+        password2 = attrs.pop('password2', None)
+        if password != password2:
+            raise serializers.ValidationError({"message": "Passwords do not match"}, password)
+        
+        return attrs
+    
+    def save(self, user):
+        password = self.validated_data.get('password')
+
+        if user.check_password(password):
+            raise serializers.ValidationError({"message": "The new password cannot be the same as the old password."}, password)
+
+        user.set_password(password)
+        user.save()
+        CustomMail.send_change_password_email(user=user.first_name, user_email=user.email, subject='Password change confirmation', to_email=user.email)
+        return user.email
+
+    class Meta:
+        model = User
+        fields = ['old_password', 'password', 'password2']
+
+
+class VerifyEmailSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(write_only=True)
+
+    def get_user_and_send_email(self, **validated_data):
+        email = self.validated_data.get('email')
+        user = User.objects.get(email=email, is_deleted=0)
+
+        # Verify Email and Return Token and id that will be used to PasswordReset
+        if user:
+            reset_token = PasswordResetTokenGenerator().make_token(user)
+            uid =  urlsafe_base64_encode(force_bytes(user.id))
+            reset_link = f"http://127.0.0.1:8000/gene2phenotype/api/reset_password/{uid}/{reset_token}"
```

Comment:
This shouldn't be hardcoded `http://127.0.0.1:8000/`

Reply:
Good spot! this has been fixed now. Thanks @dglemos